lib/pages/timetable/service/google_calendar_services.dart

```diff
@@ -0,0 +1,172 @@
+import 'package:acs_upb_mobile/resources/utils.dart';
+import 'package:googleapis/calendar/v3.dart' as g_cal;
+import 'package:googleapis/calendar/v3.dart';
+import 'package:googleapis_auth/auth_io.dart';
+
+class GoogleCalendarServices {
+  GoogleCalendarServices();
+
+  // allows us to see, edit, share, and permanently delete all the calendars you can access using GCal
+  static const List<String> _scopes = [CalendarApi.calendarScope];
+
+  static List<String> get scopes => _scopes;
+
+  // Our project IDs, used to identify an app to Google's OAuth servers.
+  static ClientId get credentials {
+    String _clientIdString;
+    if (Platform.isAndroid) {
+      _clientIdString =
+          '611150208061-4ftun8ln4v9hm1mocqs1vqcftaanj8sj.apps.googleusercontent.com';
+    } else if (Platform.isIOS) {
+      _clientIdString =
+          '611150208061-4ftun8ln4v9hm1mocqs1vqcftaanj8sj.apps.googleusercontent.com';
+    } else {
+      _clientIdString =
+          '611150208061-ljqdu5mfmjisdi1h3ics3l2sirvtpljk.apps.googleusercontent.com';
+    }
+    return ClientId(_clientIdString, '');
+  }
+}
+
+extension UniEventProviderGoogleCalendar on UniEventProvider {
+  g_cal.Event convertEvent(UniEvent uniEvent) {
+    final g_cal.Event googleCalendarEvent = g_cal.Event();
+
+    final g_cal.EventDateTime start = g_cal.EventDateTime();
+    final DateTime startDateTime = uniEvent.start.toDateTimeLocal();
+
+    start
+      ..timeZone = 'Europe/Bucharest'
+      // Google Calendar uses the IANA timezone format, but the native Dart `DateTime` uses an abbreviation provided by the operating system.
+      ..dateTime = startDateTime;
+
+    final Duration duration = uniEvent.duration.toDuration();
+
+    final g_cal.EventDateTime end = g_cal.EventDateTime();
+    final DateTime endDateTime = startDateTime.add(duration);
+    end
+      ..timeZone = 'Europe/Bucharest'
+      ..dateTime = endDateTime;
+
+    final ClassHeader classHeader = uniEvent.classHeader;
+
+    googleCalendarEvent
+      ..start = start
+      ..end = end
+      ..summary = classHeader.acronym
+      ..colorId = (uniEvent.type.googleCalendarColor.index).toString()
+      ..location = uniEvent.location;
+
+    if (uniEvent is RecurringUniEvent) {
+      final String rruleBasedOnCalendarString = uniEvent.rruleBasedOnCalendar
+          .toString()
+          .replaceAll(RegExp(r'T000000'), 'T000000Z');
+      googleCalendarEvent.recurrence = [rruleBasedOnCalendarString];
+    }
+
+    return googleCalendarEvent;
+  }
+
+  // This opens a browser window asking the user to authenticate and allow access to edit their calendar
+  Future<void> insertGoogleEvents(
+      List<g_cal.Event> googleCalendarEvents) async {
+    AutoRefreshingAuthClient client;
+    try {
+      client = await clientViaUserConsent(GoogleCalendarServices.credentials,
+          GoogleCalendarServices.scopes, Utils.launchURL);
+    } catch (e) {
+      print('Error $e in opening client.');
```

Comment:
You shouldn't continue if there was an error. Add a return here.

Also, it's good to show some kind of feedback to the user if this fails, otherwise they'll just think that the button doesn't do anything. Use AppToast.show(S.current.errorSomethingWentWrong) or some custom error in each catch block (or just wrap the entire thing in a try-catch instead of having individual blocks, your call).